### DISCUSSION TITLE
Binding to ObservableCollection

### DISCUSSION THREAD

**Discussion by TheRealMisterP** [dotnet/maui#9198] [Q&A]
Perhaps I've always done this the wrong way, but most times when I am using an ObservableCollection based on a List of the same type, I just create a new ObservableCollection using the List. But it seems recently that this method has broken the Binding for list controls (Picker is where I first noticed it).

This is a quick and dirty example that shows a little of what I am seeing. (project attached below)

MainPage code behind

`public partial class MainPage : ContentPage
{

    public MainPage()
    {
        InitializeComponent();

        var list = new List<string>
        {
            "Test 1",
            "Test 2",
            "Test 3"
        };

        // this seems to break the binding
        TestList1 = new ObservableCollection<string>(list);

        // this works
        foreach (var item in list)
        {
            TestList2.Add(item);
        }

    }

    public ObservableCollection<string> TestList1 { get; set; } = new ObservableCollection<string>();
    public ObservableCollection<string> TestList2 { get; set; } = new ObservableCollection<string>();
}
`

MainPage

`<?xml version="1.0" encoding="utf-8" ?>
<ContentPage xmlns="http://schemas.microsoft.com/dotnet/2021/maui"
             xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
             x:Name="page"
             x:Class="PickerTest.MainPage"
             BindingContext="{x:Reference page}">    
    
    <ScrollView>
        <VerticalStackLayout
            Spacing="25"
            Padding="30,0"
            VerticalOptions="Center">

            <Label Text="Binding List 1" HorizontalOptions="Center"/>
            <Picker HorizontalOptions="Center" ItemsSource="{Binding TestList1}" BackgroundColor="Gray"/>

            <Label Text="Binding List 2" HorizontalOptions="Center"/>
            <Picker HorizontalOptions="Center" ItemsSource="{Binding TestList2}"/>

            <Label Text="Binding List 1" HorizontalOptions="Center"/>
            <CollectionView ItemsSource="{Binding TestList1}" HorizontalOptions="Center" BackgroundColor="Gray"/>

            <Label Text="Binding List 2" HorizontalOptions="Center"/>
            <CollectionView ItemsSource="{Binding TestList2}" HorizontalOptions="Center"/>

        </VerticalStackLayout>
    </ScrollView>

</ContentPage>`

The controls that use TestList1 are always empty.

![image](https://user-images.githubusercontent.com/44373331/182925614-5518bd48-1bc2-4a93-a66b-21c261fe53c2.png)

Some dotnet info.

![image](https://user-images.githubusercontent.com/44373331/182930502-3ef04b91-5945-4d4d-88be-4281f0db95bd.png)

![image](https://user-images.githubusercontent.com/44373331/182930541-c55896da-befc-42aa-8515-25de238a0948.png)

So, the question seems to be: is this a bug that showed up in the latest release, or am I doing something wrong?

[PickerTest.zip](https://github.com/dotnet/maui/files/9262547/PickerTest.zip)

**Comment by sjordanGSS** [comment]
The property TestList1 doesn't fire INotifyPropertyChanged when it is assigned a new value, so the view won't be notified and will keep the binding to the old list instead of picking up the new one. That's why adding items to TestList2 (which will invoke INotifyCollectionChanged) works, but reassigning TestList1 doesn't.

**Reply by TheRealMisterP** [reply]
Thanks for the reply. I knew that when I simplified the repro I would mess something up. I am using the Community Toolkit MVVM to handle the property change notification. I'll see if I can make a better repro.
